src/localize/LocaleMacro.hx

```diff
@@ -43,8 +47,16 @@ class LocaleMacro {
             pos: Context.currentPos(),
             kind: FFun({
                 ret: null,
-                args: [{name: "catalog", type: macro : Catalog}],
-                expr: macro super(catalog)
+                args: [
+                    {name: "catalog", type: macro : Catalog},
+                    {name: "pos", type: macro : haxe.PosInfos, opt: true},
+                ],
+                expr: macro {
+                    // TODO: can we do the checking at compile-time?
```

Comment:
My general way of doing stuff is always move the checks to compile-time if possible, and I didn't add anything here as well, because if I'd use this project in a real work, I'd write a macro that checks the catalog source file against given locale class, so this check isn't needed.